cdap-api/src/main/java/co/cask/cdap/api/app/AbstractApplication.java

```diff
@@ -230,19 +232,19 @@ protected void scheduleWorkflow(Schedule schedule, String workflowName) {
   }
 
   /**
-   * Schedule the specified {@link Workflow}
+   * Schedules the specified {@link Workflow} using a time-based schedule.
    * @param scheduleName the name of the Schedule
    * @param cronTab the crontab entry for the Schedule
    * @param workflowName the name of the Workflow
    */
   protected void scheduleWorkflow(String scheduleName, String cronTab, String workflowName) {
     String scheduleDescription = scheduleName + " with crontab " + cronTab;
-    scheduleWorkflow(new Schedule(scheduleName, scheduleDescription, cronTab), workflowName,
+    scheduleWorkflow(Schedule.buildTimeSchedule(scheduleName, scheduleDescription, cronTab), workflowName,
```

Comment:
I think it's still backward compatible as long as the constructor for `Schedule` is still there and the `buildTimeSchedule` method returns an instance of Schedule (can be sub-class).